h11/_state.py

```diff
@@ -129,26 +130,70 @@
     "ERROR",
 ]
 
-CLIENT = make_sentinel("CLIENT")
-SERVER = make_sentinel("SERVER")
+
+class CLIENT(Sentinel, metaclass=Sentinel):
+    pass
```

Comment:
I'm surprised that `metaclass=...` is needed here.

(I guess I would have assumed that inheriting from a class with a custom `__new__` method was sufficient.)

Not sure this is an actionable comment, but just "this is a bit fiddly, innit?" 🙂

Reply:
Yea it is. There is some extra discussion [here](https://github.com/python-hyper/h11/pull/135#discussion_r712092061).